test-case/check-audio-equalizer.sh

```diff
@@ -52,95 +52,95 @@ func_test_eq()
         dloge "Equalizer setting failure with $conf"
         return 1
     fi
-    # test with aplay
-    dlogc "aplay -D$dev -f $fmt -c $channel -r $rate /dev/zero -d $duration"
-    aplay -D$dev -f $fmt -c $channel /dev/zero -d $duration
+
+    dlogc "$cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file"
+    $cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file
     if [ $? -ne 0 ]; then
         dloge "Equalizer test failure with $conf"
         return 1
     fi
     sleep 1
 }
 
+# this function performs IIR/FIR filter test
+# param1 must be must be iir or fir
+func_test_filter()
+{
+    local testfilter=$1
+    dlogi "1. Get amixer control id for $testfilter"
+    Filterid=`amixer -D hw:$sofcard controls | grep -i eq${testfilter}| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
+    if [ -z "$Filterid" ]; then
+        dloge "can't find $testfilter"
+        exit 1
+    fi
+
+    declare -a FilterList=($(ls -d ${current_dir}/eqctl/eq_${testfilter}_*.txt))
+    nFilterList=${#FilterList[@]}
+    dlogi "2. $testfilter list $nFilterList ${FilterList[*]}"
+    if [ $nFilterList ==  0 ]; then
+        dloge "$testfilter flter coeff list error!"
+        exit 1
+    fi
```

Comment:
normally lost configure file, it is couldn't to run `exit 2`

Reply:
In this case also same with above. normally missing configuration file can be skipped but in this case, coef files have to be existed before test. This is error case.